classes/external/load_packages.php

```diff
@@ -60,17 +61,28 @@ public static function execute(): array {
 
         $transaction = $DB->start_delegated_transaction();
 
-        // Remove every package version.
-        $versions = package_version::get_many();
-        foreach ($versions as $version) {
-            $version->delete();
-        }
-
         // Load and store packages from the application server.
         $api = new api();
         $packages = $api->get_packages();
+        $incomingpackageids = [];
         foreach ($packages as $package) {
-            $package->store();
+            [$incomingpackageids[], ] = $package->upsert();
+        }
+
+        // Remove old packages.
+        if (empty($incomingpackageids)) {
+            // There are no incoming packages -> remove every package.
+            $packages = package::get_records();
+            foreach ($packages as $package) {
+                $package->delete();
+            }
+        } else {
+            [$sql, $params] = $DB->get_in_or_equal($incomingpackageids, SQL_PARAMS_NAMED, 'packageid', false);
```

Comment:
PHP kann inzwischen Argumente per Namen übergeben :tada: 
Ich denke, das würde hier der Lesbarkeit gut tun

```suggestion
            [$sql, $params] = $DB->get_in_or_equal($incomingpackageids, SQL_PARAMS_NAMED, prefix: 'packageid', equal: false);
```